src/main/kotlin/org/rust/cargo/runconfig/test/CargoTestEventsConverter.kt

```diff
@@ -0,0 +1,203 @@
+package org.rust.cargo.runconfig.test
+
+import com.intellij.execution.process.ProcessOutputTypes
+import com.intellij.execution.testframework.TestConsoleProperties
+import com.intellij.execution.testframework.sm.ServiceMessageBuilder
+import com.intellij.execution.testframework.sm.runner.OutputToGeneralTestEventsConverter
+import com.intellij.openapi.util.Key
+import jetbrains.buildServer.messages.serviceMessages.ServiceMessage
+import jetbrains.buildServer.messages.serviceMessages.ServiceMessageVisitor
+
+/**
+ * Translate Cargo Test output to TeamCity Service Messages which SM Test Runner understands.
+ *
+ * Unfortunately, as for now Cargo Test only supports `human` message format, which is pretty
+ * obscure to handle programmatically. First of all, it is split into some separate sections,
+ * which are not easily distinguishable, that's why we do state machines here. Next problem is,
+ * that Cargo collects failed tests stdouts and print them yet at the end of test session.
+ * Lastly, we do not have exact information when given test started.
```

Comment:
>So I think extending it shouldn't be a huge problem, especially as not only we would benefit from it.

The problem would be to agree on the format I think, it's probably worth an RFC. A nice prior art is https://github.com/dart-lang/test/blob/master/doc/json_reporter.md